polygeist/tools/cgeist/driver.cc

```diff
@@ -927,6 +927,15 @@ static bool hasDeviceFuncs(mlir::gpu::GPUModuleOp deviceModule) {
   return !deviceModule.getRegion().getOps<mlir::gpu::GPUFuncOp>().empty();
 }
 
+static void eraseHostCode(mlir::ModuleOp module) {
```

Comment:
IIUC, only SYCL kernel functions are consider as device code, what happens to SYCL device functions or constructors called? Are they considered as host code and got erased right away?

Reply:
I saw them being generated in the GPU module if they are called from a SYCL kernel, with the exception of the scenario I described in [my comment](https://github.com/intel/llvm/pull/6917#pullrequestreview-1125692181)